lib/saml_idp/xml_security.rb

```diff
@@ -255,28 +221,21 @@ def validate_doc_embedded_signature(
           end
         end
 
-        base64_signature = REXML::XPath.first(
-          @sig_element,
-          '//ds:SignatureValue',
-          sig_namespace_hash
-        ).text
-
+        base64_signature = sig_element.at_xpath('//*:SignatureValue').text
         signature = Base64.decode64(base64_signature)
-        sig_alg = REXML::XPath.first(
-          signed_info_element,
-          '//ds:SignatureMethod',
-          sig_namespace_hash
-        )
+
+        sig_alg = sig_element.at_xpath('//*:SignatureMethod')
 
         log '***** validate_doc_embedded_signature: verify_signature:'
         verify_signature(base64_cert, sig_alg, signature, canon_string, soft)
       end
 
-      def digest_method_algorithm(ref, hash, digest_method_fix_enabled)
-        if digest_method_fix_enabled
-          algorithm(REXML::XPath.first(ref, '//ds:DigestMethod', hash))
+      def digest_method_algorithm(ref, digest_method_fix_enabled)
```

Comment:
this method is not long for this world, we are attempting to reach out to the one partner who will be affected by fixing this bug totally